### PR TITLE
Migrate away from Vercel

### LINKED ISSUE #414: Migrate off of Vercel
https://github.com/PaperMC/infra/issues/28
```[tasklist]
### Tasks
- [ ] https://github.com/PaperMC/docs/pull/415
```

### PR DESCRIPTION
**Pull request by powercasgamer**
Closes #414

### CODE REVIEW

File: .github/workflows/deploy.yaml [hunk 1]
```diff
@@ -24,6 +24,10 @@ jobs:
         run: "pnpm install"
       - name: "build"
         run: "pnpm run build"
+      - name: "copy files to build directory"
```
Comment (kashike) [inline .github/workflows/deploy.yaml:27]:
```suggestion
      - name: "copy cloudflare configuration files to build directory"
```

**Review by kashike — CHANGES_REQUESTED** [review]

File: .github/workflows/deploy.yaml [hunk 2]
```diff
@@ -10,7 +10,7 @@ jobs:
       deployments: write
       pull-requests: write
     steps:
-      - name: "add comment"
+      - name: Add Comment
```
Comment (zlataovce) [inline .github/workflows/deploy.yaml:13]:
```suggestion
      - name: Add comment
```

File: .github/workflows/deploy.yaml [hunk 3]
```diff
@@ -23,32 +23,43 @@ jobs:
...
-        id: "cloudflare-publish"
-        uses: "cloudflare/pages-action@v1"
+
+      - name: Install dependencies
+        run: pnpm install
+
+      - name: Build
+        run: pnpm run build
+
+      - name: Copy CloudFlare configuration files to build directory
```
Comment (zlataovce) [inline .github/workflows/deploy.yaml:47]:
```suggestion
      - name: Copy Cloudflare configuration files to build directory
```

File: .github/workflows/deploy.yaml (hunk 2 above)
Comment (kashike) [reply .github/workflows/deploy.yaml:13]: no

File: .github/workflows/deploy.yaml [hunk 4]
```diff
@@ -23,32 +23,35 @@ jobs:
             |---------------------------------|------------------------|
             | Status: | 🔨  Building... |
             ---
-      - name: "checkout"
-        uses: "actions/checkout@v4"
+      - name: Checkout repository
```
Comment (kashike) [inline .github/workflows/deploy.yaml:26]: can you undo these changes?

Comment (powercasgamer) [reply .github/workflows/deploy.yaml]: maybe

**Review by kashike — APPROVED** [review]
